Hugo migration

tbatard opened the pull request:
Site is migrated and seems to be working.

What works:
- Running using Docker: `make serve-docs`
- Running locally with Hugo present (see `site/README-HUGO.md`)
- Netlify is configured to test the Hugo site
- Matching same design
- /docs redirects to latest version
- Docs title generated from md content (Used a little [kotlin program](https://github.com/tbatard/markdown-to-frontmatter/blob/master/src/main.kt))
- New Document version generation (needs review/testing)

To do:
- SEO (Not currently working and therefore not needed, could be worth looking into creating it like in [this example](https://www.skcript.com/svr/perfect-seo-meta-tags-with-hugo/))
- RSS
- Is sitemap correct?
- Scripts parameter?
- Rebase with main (last rebase 08/13)

Open Questions:
- Is the commented part of the index.html under site folder necessary? About references

Additions:
- Related posts currently doesn't work on the real site, it always shows the same 2 posts. Hugo fixes it automatically.
- gh link in docs was not used therefore I removed it

jonasrosland commented:
Hi @tbatard !
Thank you for this, it looks great, we did find a few things that need to be fixed:

- /docs redirection does not work
- Some images are broken, such as the one on https://deploy-preview-2720--velero.netlify.app/docs/v1.4/how-velero-works/ that links to https://deploy-preview-2720--velero.netlify.app/docs/v1.4/how-velero-works/img/backup-process.png, when it should link to https://deploy-preview-2720--velero.netlify.app/docs/v1.4/img/backup-process.png
- The code block coloring is different depending on if you have specified a language or not, see https://deploy-preview-2720--velero.netlify.app/docs/v1.4/restic/
  - Code blocks that don't have any language specified are light grey, code blocks that have languages specified (yaml, shell) are in the darker tone with proper coloring, which looks better than the light grey - can we make the darker tone the default?
- Older versions are missing their TOC, see https://deploy-preview-2720--velero.netlify.app/docs/v0.3.0/ compared to https://velero.io/docs/v0.3.0/
- As RSS isn't implemented, can you please remove the link from the footer?

jonasrosland commented on hack/gen-docs.sh:
```diff
@@ -21,88 +21,92 @@ set -o errexit
-- ${NEW_DOCS_VERSION}
-" site/_config.yml
+    echo "[OS X] Adding latest version to versions list in $CONFIG_FILE"
+    sed -i '' "/- master/a\\
```
This should be `main`

jonasrosland replied:
There are a few more references to `master`, can you make sure they all point to `main`?

tbatard replied:
Yes, thank you for all your feedbacks! I'll get this done this afternoon

tbatard replied:
It should be fixed now

tbatard commented:
Hi @jonasrosland ,

I fixed most of the points

> Hi @tbatard !
> Thank you for this, it looks great, we did find a few things that need to be fixed:
> 
>     * /docs redirection does not work
> 
>     * Some images are broken, such as the one on https://deploy-preview-2720--velero.netlify.app/docs/v1.4/how-velero-works/ that links to https://deploy-preview-2720--velero.netlify.app/docs/v1.4/how-velero-works/img/backup-process.png, when it should link to https://deploy-preview-2720--velero.netlify.app/docs/v1.4/img/backup-process.png
> 
>     * The code block coloring is different depending on if you have specified a language or not, see https://deploy-preview-2720--velero.netlify.app/docs/v1.4/restic/
>       
>       * Code blocks that don't have any language specified are light grey, code blocks that have languages specified (yaml, shell) are in the darker tone with proper coloring, which looks better than the light grey - can we make the darker tone the default?
> 
>     * Older versions are missing their TOC, see https://deploy-preview-2720--velero.netlify.app/docs/v0.3.0/ compared to https://velero.io/docs/v0.3.0/
> 
>     * As RSS isn't implemented, can you please remove the link from the footer?

Most of these are fixed, I have a few questions/clarifications though:
- /docs seems to redirect correctly. If I go to 
- RSS seemed to be just for the blog posts with Jekyll, or maybe I missed something. If it's the case, then the RSS link should work. The website is configured to generate RSS content only for the blog posts. I fixed the link though as there was an issue if not on the root page.

jonasrosland commented:
Thanks for fixing the /docs redirect!
Only got two small things left:
- there are 2 files conflicting :
 ```site/_data/main-toc.yml
site/_data/v1-4-toc.yml```
- The RSS XML link looks to be generated based on the current page I'm on, for example if I go to https://deploy-preview-2720--velero.netlify.app/docs/v1.4/ the link to the RSS is https://deploy-preview-2720--velero.netlify.app/docs/v1.4/blog/index.xml, which is incorrect.

tbatard commented:
Hey @jonasrosland ,

Branch is rebased, and the RSS link is fixed.
I'm not super happy about the fix for the RSS link, I'd rather have something more dynamic instead of specifying the URLs in netlify.toml, but since we build the site it is necessary. Using `hugo serve` would make it dynamic and pick up the right URL automatically

ashish-amarnath commented:
@tbatard Looks like there are merge conflicts that need to be resolved in this PR.

tbatard commented:
@ashish-amarnath  Done! :)

jonasrosland commented:
This looks great @tbatard, thank you for fixing the remaining issues!

nrb commented:
@tbatard So far, testing this locally is nice and the preview is good! I'm going to give it a test run with some release steps for making new docs to double check things before I give an approval.

nrb commented:
I may have missed a step, but it looks like generating a new version broke the `latest` redirection on my local checkout.

I did the following:

```shell
git checkout -b "tbatard-hugo-migration" "main"
git pull git://github.com/tbatard/velero.git "hugo-migration"
git checkout test-v1.5-docs
git tag v1.5.0
NEW_DOCS_VERSION=v1.5 VELERO_VERSION=v1.5.0 make gen-docs
make serve-docs
```

Here's the [config.yaml](https://gist.github.com/nrb/8e760f222344f410b788abd6ea109928#file-config-yaml) after the changes.

When I navigate to the site, it gives me the `main` version warning, even on v1.5 URLs: <img width="2071" alt="Screen Shot 2020-08-11 at 4 28 45 PM" src="https://user-images.githubusercontent.com/321387/89945688-bba46000-dbef-11ea-9685-7be69aec4262.png">


I also [updated the v1.5 toc](https://gist.github.com/nrb/8e760f222344f410b788abd6ea109928#file-v1-5-toc-yaml), which wasn't reflected.

Is there a cached HTML directory I need to clear out that might help resolve things?

tbatard commented:
Thanks @nrb , I found the bug now it should generate the new docs version correctly!

nrb commented:
/rebase

nrb commented:
@tbatard Sorry, looks like some content edits created some conflicts that can't automatically be resolved via merge or rebase.

However, I was able to get around that with a test, and it does look like the bug is fixed, so once the conflicts are resolved, I'll approve this!

Thanks a ton for doing this migration!
